### PR TITLE
Replace CasWrapper on ImmudbWrapper

### PR DESCRIPTION
**Pull request by anfimovdm**

### CODE REVIEW

**Review by maccelf — APPROVED** [review]

**Review by zklevsha — APPROVED** [review]

File: git_notarize.py [hunk 1]
```diff
@@ -153,34 +198,33 @@ def get_debranded_imports_tag(self, tag: str):
...
         metadata['upstream_commit_sbom_hash'] = upstream_commit_sbom_hash
 
-    notarized_hash = cw.notarize(
-        local_path=f"git://{repo_path}",
-        metadata=metadata
+    result = immudb_wrapper.notarize_git_repo(
+        repo_path=repo_path,
+        user_metadata=metadata,
     )
+    notarized_hash = result.get('value', {}).get('Hash')
```
Comment (Korulag) [inline git_notarize.py:227]: Do you return the data in the same format as CAS CLI returns?

Comment (anfimovdm) [reply git_notarize.py:227]: Yes, here is an example of `immudb_wrapper` output
```
In [3]: cli.authenticate('b725b285db9d4871b9aac848cf5a4e38bbb228acf33c10dffb4adfae54f7a6a1')
Out[3]: 
{'id': 8,
 'key': 'b725b285db9d4871b9aac848cf5a4e38bbb228acf33c10dffb4adfae54f7a6a1',
 'value': {'Name': 'chan-0.0.4-3.el8.src.rpm',
  'Kind': 'file',
  'Size': '32.96 KB',
  'Hash': 'b725b285db9d4871b9aac848cf5a4e38bbb228acf33c10dffb4adfae54f7a6a1',
  'Signer': 'immudb',
  'Metadata': {'sbom_api_ver': '0.2',
   'build_id': 37,
   'build_host': '08e82fea6dbb',
   'build_arch': 'x86_64',
   'built_by': 'anfimovdm <anfimovdan@gmail.com>',
   'source_type': 'git',
   'git_url': 'https://git.almalinux.org/rpms/chan.git',
   'git_ref': 'c8',
   'git_commit': '08ac60d5244114259312eaca770568f47a567217',
   'name': 'chan',
   'epoch': None,
   'version': '0.0.4',
   'release': '3.el8',
   'arch': 'x86_64',
   'sourcerpm': None}},
 'timestamp': 1693302574,
 'verified': True,
 'refkey': None,
 'revision': 1}
```

File: git_notarize.py [hunk 2]
```diff
@@ -199,175 +243,209 @@ def cli_main():
...
+    immudb_username, immudb_password = (
+        args.immudb_username or os.getenv('IMMUDB_USERNAME'),
+        args.immudb_password or os.getenv('IMMUDB_PASSWORD'),
+    )
+    if not immudb_username:
         logging.error(
-            "No CAS_API_KEY found, and it's required to operate with CAS. " \
-            "You can either set it as an environmental variable (CAS_API_KEY) " \
-            "or providing it using the --cas-api-key option."
+            "No IMMUDB_USERNAME found, and it's required to operate with CAS. "
```
Comment (javihernandez) [inline git_notarize.py:252]: required to operate with IMMUDB

**Review by javihernandez — CHANGES_REQUESTED** [review]
LGTM. I still see some references to CAS around, can you please take care of them?

**Review by javihernandez — APPROVED** [review]

**Pull request merged**